### DISCUSSION TITLE
Using homebrew-installed packages during Xcode's Run Script phase

### DISCUSSION THREAD

**Discussion by smalsam-newr** [Homebrew/discussions#3035] [Everyday usage]
When Xcode runs a script during the build's Run Script phase, it has its own environment this runs in. This environment's PATH includes /usr/local/bin, but not /opt/homebrew. This means that, on an Intel Mac, I could run, say, cmake (or whatever) installed from Homebrew, but it is not found on an Apple Silicon Mac. What is the preferred fix for this, such that I could continue to run a script on both Intel and Apple Silicon platforms?

**Comment by SMillerDev** [comment]
> What is the preferred fix for this, such that I could continue to run a script on both Intel and Apple Silicon platforms?

Whatever fix you prefer. I would personally check if you're running on ARM and just call the binary directly depending on that.

**Reply by smalsam-newr** [reply]
The problem with that suggestion, is that there are other things that are run as part of CMake, which are not called by full path, however, since they are run under the XCode environment, they don't have Homebrew's path, and thus cannot be called.

**Reply by SMillerDev** [reply]
That wasn't in the original question but my answer is easy enough to adapt. If you need something in `PATH` you can just add both locations to `PATH` and it'll pick the first one it finds.

**Comment by iosdevben** [comment]
I have `/opt/homebrew/bin` in my `$PATH` in `.bashrc` and `.profile` but they are missing from `$PATH` when the run scripts run. They are invoked as `/bin/sh`, eg:

![image](https://user-images.githubusercontent.com/824608/174906003-a8a137ba-ad8e-44b0-b4d0-3ab2ea4a26c6.png)

**Reply by asam139** [reply]
Let's add 

unset SDKROOT
export PATH="$PATH:/opt/homebrew/bin"
